nhrpd/nhrp_vty.c

```diff
@@ -1155,6 +1205,12 @@ static int interface_config_write(struct vty *vty)
 		if (nifp->source)
 			vty_out(vty, " tunnel source %s\n", nifp->source);
 
+		if (nifp->auth_token) {
+			auth = (struct nhrp_cisco_authentication_extension *)
```

Comment:
nit: typecasting is not needed (same type as left operand).

Reply:
doesn't seem to be the case
auth             | nhrp_cisco_authentication_extension
auth_token | zbuf*
buf               | uint8_t*